### PR TITLE
Ensuring that the ScriptHost is Disposed of in case of an exception in the run loop.

### LINKED ISSUE #72: ScriptHost is not disposed if an exception occurs in run loop
The ScriptHost is not disposed if an exception occurs in the run loop in the ScriptHostManager.

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #72

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHostManager.cs [hunk 1]
```diff
@@ -108,6 +116,20 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
                         _traceWriter.Error("A ScriptHost error occurred", ex);
                     }
 
+                    // If a ScriptHost instance was created before the exception was thrown
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHostManager.cs]: this "before the exception was thrown" comment confuses me. What exception?

File: src/WebJobs.Script/Host/ScriptHostManager.cs [hunk 2]
```diff
@@ -133,20 +155,26 @@ private static void LogErrors(ScriptHost host)
         }
 
         // Let the existing host instance finish currently executing functions.
-        private async Task Orphan(ScriptHost instance)
+        private async Task Orphan(ScriptHost instance, bool forceStop = false)
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHostManager.cs]: Add a descriptive comment block on this method, explaining the "forceStop" param

File: test/WebJobs.Script.Tests/ScriptHostManagerTests.cs [hunk 3]
```diff
@@ -59,6 +61,34 @@ public class ScriptHostManagerTests
             }
         }
 
+        [Fact]
+        public void RunAndBlock_DisposesOfHost_WhenExceptionIsThrown()
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/ScriptHostManagerTests.cs]: Thanks for adding test. We need to get in the habit of this for all new code now, to stop the debt :)

**Comment by mathewc** [comment]
Looks good - just address my few minor comments

File: src/WebJobs.Script/Host/ScriptHostManager.cs [hunk 4]
```diff
@@ -132,21 +154,33 @@ private static void LogErrors(ScriptHost host)
             }
         }
 
-        // Let the existing host instance finish currently executing functions.
-        private async Task Orphan(ScriptHost instance)
+        /// <summary>
```
Comment (fabiocav) [inline src/WebJobs.Script/Host/ScriptHostManager.cs]: @mathewc can you take a look at this comment doc and let me know if you're happy with the verbiage there?

**Pull request merged**